pkg/stores/proxy/unformatter.go

```diff
@@ -0,0 +1,56 @@
+package proxy
+
+import (
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/wrangler/pkg/data"
+)
+
+// unformatterStore removes fields added by the formatter that kubernetes cannot recognize.
+type unformatterStore struct {
+	types.Store
+}
+
+// ByID looks up a single object by its ID.
+func (u *unformatterStore) ByID(apiOp *types.APIRequest, schema *types.APISchema, id string) (types.APIObject, error) {
+	return u.Store.ByID(apiOp, schema, id)
+}
+
+// List returns a list of resources.
+func (u *unformatterStore) List(apiOp *types.APIRequest, schema *types.APISchema) (types.APIObjectList, error) {
+	return u.Store.List(apiOp, schema)
+}
+
+// Create creates a single object in the store.
+func (u *unformatterStore) Create(apiOp *types.APIRequest, schema *types.APISchema, data types.APIObject) (types.APIObject, error) {
+	return u.Store.Create(apiOp, schema, data)
+}
+
+// Update updates a single object in the store.
+func (u *unformatterStore) Update(apiOp *types.APIRequest, schema *types.APISchema, data types.APIObject, id string) (types.APIObject, error) {
+	data = unformat(data)
+	return u.Store.Update(apiOp, schema, data, id)
+}
+
+// Delete deletes an object from a store.
+func (u *unformatterStore) Delete(apiOp *types.APIRequest, schema *types.APISchema, id string) (types.APIObject, error) {
+	return u.Store.Delete(apiOp, schema, id)
+
+}
+
+// Watch returns a channel of events for a list or resource.
+func (u *unformatterStore) Watch(apiOp *types.APIRequest, schema *types.APISchema, wr types.WatchRequest) (chan types.APIEvent, error) {
+	return u.Store.Watch(apiOp, schema, wr)
+}
+
+func unformat(obj types.APIObject) types.APIObject {
+	unst, ok := obj.Object.(map[string]interface{})
+	if !ok {
+		return obj
+	}
+	data.RemoveValue(unst, "metadata", "fields")
+	data.RemoveValue(unst, "metadata", "relationships")
+	data.RemoveValue(unst, "metadata", "state")
+	data.RemoveValue(unst, "status") // client-go UpdateStatus is not supported
```

Comment:
I was under the impression that the Update call from [ResourceInterface](https://pkg.go.dev/k8s.io/client-go/dynamic#ResourceInterface) includes the Status since it is a part of the object. And the UpdateStatus call is used to update the status only. 

If this is not true and Update calls to K8s will not update the status even if it differs from the stored object then this should be dropped.

Reply:
I don't think that's the case. Here's what the [docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#status-subresource) say:

> PUT/POST/PATCH requests to the custom resource ignore changes to the status stanza.

Here's a brief reproducer showing this: https://gist.github.com/cmurphy/1512fa9482f8eba5cfed352f5d809894

Even though the status isn't updated, the API server still validates its fields, which is why we see the warnings about .status.conditions in the response in steve. So the status object still needs to be valid even though it's not being used.

Reply:
Thanks for the example! It was easy to run and worked exactly as you said. I didn't know about that behavior before this. Though I did notice that the behavior depends on `status` being defined as a subresource in the CRD definition. If I removed the `status` subresource line in `crds.yaml`, then changes to that status field persisted which concerns me because some of our resources such as `v3.User` do not have `status` defined as a subresource`
``` yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  creationTimestamp: "2023-06-16T14:15:37Z"
  generation: 1
  name: users.management.cattle.io
  resourceVersion: "743"
  uid: 06aa895c-bfbf-4f71-8a4f-43e31cd2bfcd
spec:
  conversion:
    strategy: None
  group: management.cattle.io
  names:
    kind: User
    listKind: UserList
    plural: users
    singular: user
  scope: Cluster
  versions:
  - name: v3
    schema:
      openAPIV3Schema:
        type: object
        x-kubernetes-preserve-unknown-fields: true
    served: true
    storage: true
   ```

Reply:
Thanks, that's unfortunate that our CRDs don't conform to k8s standards. I'll rework this to drop the individual condition fields in the condition slice.

Reply:
We may be able to fix most of these with our crd generation changes.